pkg/storage/stores/tsdb/store.go

```diff
@@ -32,78 +33,64 @@ type store struct {
 	indexShipper      indexshipper.IndexShipper
 	indexWriter       IndexWriter
 	backupIndexWriter index.Writer
+	logger            log.Logger
 	stopOnce          sync.Once
 }
 
-var storeInstance *store
-
-// This must only be called in test cases where a new store instances
-// cannot be explicitly created.
-func ResetStoreInstance() {
-	if storeInstance == nil {
-		return
-	}
-	storeInstance.Stop()
-	storeInstance = nil
-}
-
 type newStoreFactoryFunc func(
+	name string,
 	indexShipperCfg indexshipper.Config,
-	p config.PeriodConfig,
+	schemaCfg config.SchemaConfig,
 	f *fetcher.Fetcher,
 	objectClient client.ObjectClient,
 	limits downloads.Limits,
-	tableRanges config.TableRanges,
+	tableRange config.TableRange,
 	backupIndexWriter index.Writer,
 	reg prometheus.Registerer,
+	logger log.Logger,
 ) (
 	indexReaderWriter index.ReaderWriter,
 	stopFunc func(),
 	err error,
 )
 
-// NewStore creates a new store if not initialized already.
-// Each call to NewStore will always build a new stores.ChunkWriter even if the store was already initialized since
-// fetcher.Fetcher instances could be different due to periodic configs having different types of object storage configured
-// for storing chunks.
-// It also helps us make tsdb store a singleton because
-// we do not need to build store for each schema config since we do not do any schema specific handling yet.
-// If we do need to do schema specific handling, it would be a good idea to abstract away the handling since
-// running multiple head managers would be complicated and wasteful.
+// NewStore creates a new tsdb index ReaderWriter.
 var NewStore = func() newStoreFactoryFunc {
```

Comment:
`newStoreFactoryFunc` was just a hack to make tsdb store instances a singleton. I think we can get rid of it.